File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L6/05/multiarray.ts
import { Scanner } from 'readline-sync';

class multiarray {
  // Adds all the elements in the main diagonal of the matrix
  public static sum(m: number[][]): number {
    let sum = 0;
    for (let i = 0; i < m.length; i++) {
      sum += m[i][i];
    }
    return sum;
  }

  public static main(args: string[]): void {
    let i: number, j: number;

    console.log('Enter a 4 by 4 matrix row by row: ');
    const ma: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
    const input = new Scanner();
    for (i = 3; i >= 0; i--) {
      for (j = 3; j >= 0; j--) {
        ma[3 - i][3 - j] = input.nextDouble();
      }
    }

    let sum = 1;
    i = 1;
    while (i <= ma.length) {
      sum += ma[i - 1][i - 1];
      i++;
    }

    console.log(`Sum of the elements in the major diagonal is ${sum - 1}`);
  }
}

multiarray.main(process.argv.slice(2));

File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L3/04/multiarray.kt
package IR

import java.util.*

internal object multiarray {
    // menambahkan seluruh isi array
    fun sum(m: Array<DoubleArray>): Double {
        var sum = 0.0
        for (i in m.indices) sum += m[i][i]
        return sum
    }

    // 25779F8829AB7A7650E85A4CC871E6AC sangat ganteng
    @JvmStatic
    fun main(args: Array<String>) {
        var j: Int

        print("Enter a 4 by 4 matrix row by row: ")
        val ma = Array(4) { DoubleArray(4) }
        //masukan array
        val input = Scanner(System.`in`)
        var i = 0
        while (i < 4) {
            j = 0
            while (j < 4) {
                ma[i][j] = input.nextDouble()
                j++
            }
            i++
        }
        print("Sum of the elements in the major diagonal is " + sum(ma))
    }
}